### PR TITLE
Support URI::File of Ruby 2.6

### PR DESCRIPTION
**Pull request by hsbt**
### What was the end-user problem that led to this PR?

When users use file protocol with bundler, URI of Ruby 2.6 will raise auth info assignment and normalize host parameter given localhost variable with its case.

### Why did you choose this fix out of the possible options?

URI behavior of Ruby 2.6 is reasonable for me. Because I will fix this problem on bundler side.

### CODE REVIEW

**Comment by hsbt** [comment]
I found the normalization of `localhost` params with Ruby 2.6. I try to fix it for the old ruby versions.

File: spec/support/matchers.rb [hunk 1]
```diff
@@ -236,6 +236,7 @@ def plugin_should_not_be_installed(*names)
     end
 
     def lockfile_should_be(expected)
+      expected = expected.dup.gsub!(%r{file:\/\/localhost}, "file://") if defined?(URI::File)
```
Comment (segiddins) [inline spec/support/matchers.rb]: why is this change necessary?

**Comment by segiddins** [comment]
Looks like there are still a bunch of failures on 2.6, not sure if that's expected? https://travis-ci.org/bundler/bundler/jobs/392208030

File: spec/support/matchers.rb (hunk 1 above)
Comment (hsbt) [reply spec/support/matchers.rb]: `URI::File` normalize `localhost` params with https://github.com/ruby/ruby/blob/trunk/lib/uri/file.rb#L57

Comment (segiddins) [reply spec/support/matchers.rb]: Is that new in 2.6? If so, mind adding a comment ?

**Comment by hsbt** [comment]
@segiddins I update latest commits. For now, All of the versions of Ruby works fine. Can you review again?

File: spec/commands/exec_spec.rb [hunk 2]
```diff
@@ -789,6 +789,8 @@ def bin_path(a,b,c)
       end
 
       it "overrides disable_shared_gems so bundler can be found" do
+        skip "bundler 1.16.x is not support with Ruby 2.6 on Travis CI" if RUBY_VERSION >= "2.6"
```
[the remark this answers is not in the document]
Comment (hsbt) [reply spec/commands/exec_spec.rb]: System bundler is not contained this pull request. So, It always raises `URI::File` with Ruby 2.6. After v1.16.3 contained this fix was released, We can remove this.

**Comment by segiddins** [comment]
@bundlerbot r+
